### PR TITLE
Refactored code and fixed textarea event handling

### LINKED ISSUE #78: Navigating editor with arrow keys jumps/scrolls the editor to make the current cursor position the last line of the editor.

### PR DESCRIPTION
**Pull request by brianlee389**
Some code has been condensed. Old and unneeded comments have been removed. Fixes #78 and removing unnecessary all to refreshEditors function.

### CODE REVIEW

File: snowball-template.php [hunk 1]
```diff
@@ -30,4 +30,4 @@
 
   </article>
 </body>
-</html>
+</html>
```
Comment (thomaspark) [inline snowball-template.php]: Can avoid making any changes to this file by adding back the newline here.

File: snowball.php [hunk 2]
```diff
@@ -376,11 +373,13 @@ function snowball_save_block($json_block, $post_id) {
...
 }
 
-// if successful return insert_id
-// if fails return -1
-// TODO: error checking beginning and after the function.
 function snowball_save_article($article, $post_id) {
   global $wpdb;
+  $nonce = $_POST['snowball_ajax_nonce'];
+
+  if (! wp_verify_nonce( $nonce, 'snowball_ajax_nonce')) {
```
Comment (thomaspark) [inline snowball.php]: Let's delete the space after `!`. Not sure if it makes a difference, but just to be safe & consistent.

Also can delete space before `$nonce`.

File: snowball.php [hunk 3]
```diff
@@ -376,11 +373,13 @@ function snowball_save_block($json_block, $post_id) {
...
+    die ( 'Busted!');
```
Comment (thomaspark) [inline snowball.php]: `die('Busted!');`

**Comment by thomaspark** [comment]
Nonce checking is nice. Change from `keyup` to `input` is good too. Only downside is no IE8 support, and IE9 is buggy when deleting characters, but I think it's worth it.

I like the goal of combining some of these functions, but suggest a different approach: `renderPreview` currently takes the widget values and updates preview. `renderBlockWithEditor` takes codemirror values and updates preview. So how about we combine them by having `renderPreview` also grab the codemirror values before refreshing the preview? Makes sense for `renderPreview` to update preview based on all values, not just from widgets.

I saw that you conditionally call `refreshEditors` by checking `e.currentTarget.parentNode.parentNode.getAttribute("class");`. Instead of navigating the dom like this (which could break if we change the markup), let's create two separate event handlers (one for `.snowball-tinker input, .snowball-tinker textarea`, and one for `.snowball-code textarea`) and just call `refreshEditors` in the first one.

Left a few other minor comments inline.

File: snowball.php (hunk 3 above)
Comment (brianlee389) [inline snowball.php]: I forgot to change that

File: snowball.php [hunk 4]
```diff
@@ -334,9 +327,13 @@ function snowball_install_dbtable() {
 
 // if successful return insert_id
 // if fails return -1
-// TODO: error checking beginning and after the function.
 function snowball_save_block($json_block, $post_id) {
   global $wpdb;
+  $nonce = $_POST['snowball_ajax_nonce'];
+
+  if (! wp_verify_nonce( $nonce, 'snowball_ajax_nonce')) {
```
Comment (thomaspark) [inline snowball.php]: `!wp_verify_nonce`

**Comment by thomaspark** [comment]
Just need to remove the general `change` handler, and add `change` to `snowball-tinker`'s `input` handler:

```
.on("input, change", ".snowball-tinker input, .snowball-tinker textarea"...
```

**Pull request merged**